### ISSUE TITLE
redis.exceptions.TimeoutError @ single server setup

### ISSUE THREAD

**Issue by dusatvoj** [frappe/frappe_docker#1024]
## Description of the issue
I've followed [this](https://github.com/frappe/frappe_docker/blob/main/docs/single-server-example.md) guide and ended up with error below. I can't connect to redis, even redis is running.

## Steps to reproduce the issue

1. Follow guide
2. Have issue at redis connection
3.

### Observed result
Can't connect to redis

### Expected result
Connected to redis, succesfuly initialized bench

### Stacktrace / full error message if available

```
[22:34:14] root@erpnext:~# docker compose --project-name erpnext-one exec backend bench new-site erp.one.cz --mariadb-root-password wuRB98eA2zkkJiR3pmnW28Th --install-app erpnext --admin-password K8w4B6pFAiaeTwFQLU46vqrG 
Traceback (most recent call last):
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/connection.py", line 559, in connect
    sock = self._connect()
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/connection.py", line 615, in _connect
    raise err
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/connection.py", line 603, in _connect
    sock.connect(socket_address)
TimeoutError: [Errno 110] Connection timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/patched_bench_helper.py", line 48, in <module>
    raise SystemExit(main())
  File "/usr/local/bin/patched_bench_helper.py", line 43, in main
    frappe.utils.bench_helper.main()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 18, in main
    click.Group(commands=commands)(prog_name="bench")
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/commands/site.py", line 72, in new_site
    frappe.init(site=site, new_site=True)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 248, in init
    setup_module_map()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1514, in setup_module_map
    local.app_modules = _cache.get_value("app_modules")
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/redis_wrapper.py", line 72, in get_value
    val = self.get(key)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/client.py", line 1606, in get
    return self.execute_command('GET', name)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/client.py", line 898, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/connection.py", line 1192, in get_connection
    connection.connect()
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/redis/connection.py", line 561, in connect
    raise TimeoutError("Timeout connecting to server")
redis.exceptions.TimeoutError: Timeout connecting to server
[1] [22:36:32] root@erpnext:~# docker ps
CONTAINER ID   IMAGE                             COMMAND                  CREATED          STATUS                   PORTS                                       NAMES
c821bafe9c2c   frappe/erpnext-nginx:v14.10.1     "/docker-entrypoint.…"   19 minutes ago   Up 5 minutes             0.0.0.0:8080->8080/tcp, :::8080->8080/tcp   erpnext-one-frontend-1
5f79ae450a63   frappe/erpnext-worker:v14.10.1    "bench schedule"         19 minutes ago   Up 5 minutes                                                         erpnext-one-scheduler-1
a27177bd441b   frappe/erpnext-worker:v14.10.1    "/home/frappe/frappe…"   19 minutes ago   Up 5 minutes                                                         erpnext-one-backend-1
753aee64c0ad   redis:6.2-alpine                  "docker-entrypoint.s…"   20 minutes ago   Up 5 minutes             6379/tcp                                    erpnext-one-redis-1
b4f1705c7d4e   mariadb:10.6                      "docker-entrypoint.s…"   34 minutes ago   Up 5 minutes (healthy)   3306/tcp                                    mariadb-database
fd4a664c8650   gcr.io/cadvisor/cadvisor:latest   "/usr/bin/cadvisor -…"   5 weeks ago      Up 5 minutes (healthy)   0.0.0.0:8088->8080/tcp                      cadvisor_erpnext
[130] [22:40:06] root@erpnext:~# docker logs erpnext-one-redis-1 --tail 10
1:C 18 Dec 2022 21:33:42.309 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 18 Dec 2022 21:33:42.310 * monotonic clock: POSIX clock_gettime
1:M 18 Dec 2022 21:33:42.310 * Running mode=standalone, port=6379.
1:M 18 Dec 2022 21:33:42.310 # Server initialized
1:M 18 Dec 2022 21:33:42.311 * Loading RDB produced by version 6.2.8
1:M 18 Dec 2022 21:33:42.311 * RDB age 39 seconds
1:M 18 Dec 2022 21:33:42.311 * RDB memory usage when created 0.77 Mb
1:M 18 Dec 2022 21:33:42.311 # Done loading RDB, keys loaded: 0, keys expired: 0.
1:M 18 Dec 2022 21:33:42.311 * DB loaded from disk: 0.001 seconds
1:M 18 Dec 2022 21:33:42.311 * Ready to accept connections
```

**Comment by dusatvoj** [comment]
```
[22:40:12] root@erpnext:~# docker exec -it erpnext-one-redis-1 sh
/data # netstat -pltnu
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:6379            0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.11:36165        0.0.0.0:*               LISTEN      -
tcp        0      0 :::6379                 :::*                    LISTEN      -
udp        0      0 127.0.0.11:40874        0.0.0.0:*                           -
/data # redis-
redis-benchmark  redis-check-aof  redis-check-rdb  redis-cli        redis-sentinel   redis-server
/data # redis-
redis-benchmark  redis-check-aof  redis-check-rdb  redis-cli        redis-sentinel   redis-server
/data # redis-cli 
127.0.0.1:6379> ping
PONG
```

**Comment by revant** [comment]
make sure you've correct common_site_config.json generated. it sets redis host/port.

**Comment by dusatvoj** [comment]
Where can I find this config? Can't see it anywhere and there were no info about that file at example setup.

**Comment by dusatvoj** [comment]
Oh, I found it in backend image and it looks good ....
```
{"db_host": "mariadb-database", "db_port": 3306, "redis_cache": "redis://redis:6379/0", "redis_queue": "redis://redis:6379/1", "redis_socketio": "redis://redis:6379/2", "socketio_port": 9000}
```

**Comment by revant** [comment]
there need to be 3 redis containers now. check pwd.yml

I think redis databases e.g. /0, /1, /2 is deprecated in latest redis.

**Comment by dusatvoj** [comment]
Ok, when the database identifier is deprecated, I think you have to reflect it in the app and docker composes. There's running just one redis after following the manual and running that stuff.

**Comment by revant** [comment]
> Ok, when the database identifier is deprecated, I think you have to reflect it in the app and docker composes. There's running just one redis after following the manual and running that stuff.

I don't know if it's deprecated or not recommended.
I think it is available in redis 6.2 which is tagged to `redis:6.2-alpine` in compose.redis.yaml

May be something else is wrong.
backend container or any container with worker image has `wait-for-it` command available in it. Try connecting to services like db and redis from other containers to find more about connection errors. e.g. `wait-for-it mariadb-database:3306` or `wait-for-it redis:3679`

**Comment by dusatvoj** [comment]
```
frappe@bcaf9558f48f:~/frappe-bench/sites$ wait-for-it mariadb-database:3306
wait-for-it: waiting 15 seconds for mariadb-database:3306
wait-for-it: timeout occurred after waiting 15 seconds for mariadb-database:3306
frappe@bcaf9558f48f:~/frappe-bench/sites$ wait-for-it redis:6379
wait-for-it: waiting 15 seconds for redis:6379
wait-for-it: timeout occurred after waiting 15 seconds for redis:6379
```
In docker-compose.yml:
```
...
  backend:
    depends_on:
      configurator:
        condition: service_completed_successfully
    image: frappe/erpnext-worker:v14.10.1
    networks:
      bench-network: null
      mariadb-network: null
...
networks:
  bench-network:
    name: erpnext-curso
  mariadb-network:
    name: mariadb-network
    external: true
  traefik-public:
    name: traefik-public
    external: true
...
```

```
# docker network ls 
NETWORK ID     NAME              DRIVER    SCOPE
dfdc43ea3be4   bridge            bridge    local
1511d53698aa   erpnext-curso     bridge    local
6282cd747bb8   host              host      local
6f8699e07ce3   mariadb-network   bridge    local
a07d58e6807b   none              null      local
739c5d8d9401   traefik-public    bridge    local
```

... I'm confused why there's `null` in network section of docker compose :thinking:

**Comment by dusatvoj** [comment]
I did the compose file by running:
`docker compose --project-name erpnext-one --env-file ~/gitops/erpnext.env -f compose.yaml -f overrides/compose.erpnext.yaml -f overrides/compose.redis.yaml -f docs/compose/compose.multi-bench.yaml -f overrides/compose.noproxy.yaml config >~/gitops/erpnext-one.yaml`

**Comment by dusatvoj** [comment]
@revant do you have any suggestion? :thinking: :pray:

**Comment by revant** [comment]
recently I faced such issue when the docker was installed using snap.
Started with a clean install of docker from `curl -sSL https://get.docker.com | bash` and connection worked.

**Comment by dusatvoj** [comment]
Well, I found an issue at my network stack :man_facepalming: :upside_down_face: .
Now there's an issue that compose can't be built up due to `service "redis" has neither an image nor a build context specified: invalid compose project` error. The redis service is defined at `overrides/compose.multi-bench.yaml` but it's probably thing for a new issue.

**Comment by revant** [comment]
> Well, I found an issue at my network stack 🤦‍♂️ 🙃 . Now there's an issue that compose can't be built up due to `service "redis" has neither an image nor a build context specified: invalid compose project` error. The redis service is defined at `overrides/compose.multi-bench.yaml` but it's probably thing for a new issue.

fixed that in #1042
